lib/msf/core/post/linux/compile.rb

```diff
@@ -36,7 +50,12 @@ def live_compile?
   def upload_and_compile(path, data, compiler_args='')
     write_file "#{path}.c", strip_comments(data)
 
-    compiler_cmd = "#{datastore['COMPILER']} -o '#{path}' '#{path}.c'"
+    compiler = datastore['COMPILER']
+    if datastore['COMPILER'] == 'Auto'
+      compiler = get_compiler
+    end
```

Comment:
This is where we'd likely want to fail if `#get_compiler` can't return a compiler and the module ignored/didn't call `#live_compile?`.